packages/backend/src/tenants/service.ts

```diff
@@ -0,0 +1,209 @@
+      { tenantId: id },
+      'could not find auth tenant entry for existing backend entry'
+    )
+    return undefined
+  }
+
+  const { idpConsentUrl, idpSecret } = authTenant
+  return {
+    ...tenant,
+    idpConsentUrl,
+    idpSecret
+  }
+}
+
+interface CreateTenantOptions {
+  email: string
+  apiSecret: string
+  idpSecret: string
+  idpConsentUrl: string
+  publicName?: string
+}
+
+async function createTenant(
+  deps: ServiceDependencies,
+  options: CreateTenantOptions
+): Promise<Tenant> {
+  const trx = await deps.knex.transaction()
+  try {
+    const { email, apiSecret, publicName, idpSecret, idpConsentUrl } = options
+    const tenant = await Tenant.query(trx).insertAndFetch({
+      email,
+      publicName,
+      apiSecret,
+      idpSecret,
+      idpConsentUrl
+    })
+
+    const mutation = gql`
+      mutation CreateAuthTenant($input: CreateTenantInput!) {
+        createTenant(input: $input) {
+          tenant {
+            id
+          }
+        }
+      }
+    `
+
+    const variables = {
+      input: {
+        id: tenant.id,
+        idpSecret,
+        idpConsentUrl
+      }
+    }
+
+    // TODO: add type to this in https://github.com/interledger/rafiki/issues/3125
+    await deps.apolloClient.mutate({ mutation, variables })
+    await trx.commit()
+    return tenant
+  } catch (err) {
+    await trx.rollback()
+    throw err
+  }
+}
+
+interface UpdateTenantOptions {
+  id: string
+  email?: string
+  publicName?: string
+  apiSecret?: string
+  idpConsentUrl?: string
+  idpSecret?: string
+}
+
+async function updateTenant(
+  deps: ServiceDependencies,
+  options: UpdateTenantOptions
+): Promise<Tenant> {
+  const trx = await deps.knex.transaction()
+
+  try {
+    const { id, apiSecret, email, publicName, idpConsentUrl, idpSecret } =
+      options
+    const tenant = await Tenant.query(trx)
+      .patchAndFetchById(options.id, {
+        email,
+        publicName,
+        apiSecret,
+        idpConsentUrl,
+        idpSecret
+      })
+      .whereNull('deletedAt')
+      .throwIfNotFound()
+
+    if (idpConsentUrl || idpSecret) {
+      const mutation = gql`
+        mutation UpdateAuthTenant($input: UpdateTenantInput!) {
+          updateTenant(input: $input) {
+            tenant {
+              id
+            }
+          }
+        }
+      `
+
+      const variables = {
+        input: {
+          id,
+          idpConsentUrl,
+          idpSecret
+        }
+      }
+
+      // TODO: add types to this in https://github.com/interledger/rafiki/issues/3125
+      await deps.apolloClient.mutate({ mutation, variables })
+    }
+
+    await trx.commit()
+    return tenant
+  } catch (err) {
+    await trx.rollback()
+    throw err
+  }
+}
+
+async function deleteTenant(
+  deps: ServiceDependencies,
+  id: string
+): Promise<void> {
+  const trx = await deps.knex.transaction()
+
+  try {
+    await Tenant.query(trx).patchAndFetchById(id, {
+      deletedAt: new Date(Date.now())
+    })
```

Comment:
maybe we can pass through the `deletedAt` to the mutation, so both tables in backend and auth have the same value